src/preferences.c

```diff
@@ -77,3 +77,50 @@ void SetUserPreference(const char* key, double value) {
   fclose(preferences);
   free(kSerializedJsonNewlineWithNewline);
 }
+
+double GetUserPreference(const char* key) {
+  double value = 0;
+  FILE* user_preferences = fopen("preferences.json", "r");
+
+  if (user_preferences == NULL) {
+    printf("Error opening user preferences file!\n");
+    return value;
+  }
+
+  fseek(user_preferences, 0, SEEK_END);
+  unsigned long kFileSize = (unsigned long)ftell(user_preferences);
+  fseek(user_preferences, 0, SEEK_SET);
+
+  char* kFileBuffer = (char*)calloc(kFileSize + 1, sizeof(char));
+  if (kFileBuffer == NULL) {
+    printf("Error allocating memory for file buffer!\n");
+    fclose(user_preferences);
+    return value;
+  }
+  size_t bytes_read =
+      fread(kFileBuffer, sizeof(char), kFileSize, user_preferences);
+  fclose(user_preferences);
+
+  if (bytes_read != (size_t)kFileSize) {
```

Comment:
Ok, but why cast to `size_t`?

Reply:
Redundant casts are... redundant. Try to remove it and see if CLion complains, i.e. suggests another cast. If not, you are good to go

Reply:
It seems like `unsigned long` and `size_t` are comparable:
<img width="689" alt="image" src="https://github.com/audio-engineer/aau-p1-software/assets/70742985/c575e4b7-20b9-4bd2-ae60-e772146504c5">